pkg/handlers/handlers.go

```diff
@@ -83,16 +83,32 @@ func newMimeTypeReader(r io.Reader) (mimeTypeReader, error) {
 func newFileReader(r io.Reader) (fileReader, error) {
 	var fReader fileReader
 
-	fReader.BufferedReadSeeker = iobuf.NewBufferedReaderSeeker(r)
+	// To detect the MIME type of the input data, we need a reader that supports seeking.
+	// This allows us to read the data multiple times if necessary without losing the original position.
+	// We use a BufferedReaderSeeker to wrap the original reader, enabling this functionality.
+	// If an error occurs during MIME type detection, it is important we close the BufferedReaderSeeker
+	// to release any resources it holds and prevent potential memory leaks.
+	bufferedReader := iobuf.NewBufferedReaderSeeker(r)
+	fReader.BufferedReadSeeker = bufferedReader
+
+	var err error
+	defer func() {
+		if err != nil {
+			if closeErr := bufferedReader.Close(); closeErr != nil {
+				err = fmt.Errorf("%w; error closing reader: %w", err, closeErr)
+			}
+		}
```

Comment:
Why do we only close in the error case?

Reply:
In the non-error case, the caller is responsible for closing the reader, so I should clarify that the caller generally handles this after use. In this case, `Close` only returns the buffer to the pool if MIME type detection fails.

To reduce confusion, I could change the return type to a pointer and return `nil`. Also, since we lazily fetch the buffer in `Read`, I might avoid checking out the buffer from the pool in `NewBufferedReaderSeeker`, which leaves the responsibility of closing on the caller.